qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -137,99 +142,200 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             Output dag with UnitaryGates synthesized to target basis.
 
         Raises:
-            TranspilerError:
-                1. pulse_optimize is True but pulse optimal decomposition is not known
-                   for requested basis.
-                2. pulse_optimize is True and natural_direction is True but a preferred
-                   gate direction can't be determined from the coupling map or the
-                   relative gate lengths.
+            QiskitError: if a 'method' was specified for the class and is not
```

Comment:
Is there a benefit from broadening the error from `TranspilerError` to `QiskitError`?  It's still kind of a transpilation error to be missing a transpiler plugin, though I could certainly see the argument the other way around.  I suppose broadening it means that a user can catch `TranspilerError` and still have the missing plugins propagated, but I'm not certain there's a use case for separating those?